File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    if(s[0] === ")" || s[s.length-1] === "("){
        return false;
    }
    let a = 0;
    s.split('').forEach((item,index) => {
        if(item === "("){
            a++;
        } else if(item === ")"){
            if(a>0) a--;
            else return false;
        }
    })
    return a === 0
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    
    for i in s:
        if i == '(':
            stack.append(i);
        elif i == ')':
            if len(stack) == 0:
                return False
            if stack[-1] == '(':
                stack.pop()
    
    if len(stack) == 0:
        return True
    else:
        return False